src/main/java/it/geosolutions/httpproxy/HTTPProxy.java

```diff
@@ -209,12 +209,18 @@ public HttpRoute determineRoute(
                     LOGGER.info("Returning proxy route");
                     if (target.getSchemeName().equals("http")) {
                         LOGGER.debug("Setting http scheme");
-                        return new HttpRoute(target, null, httpHost,
-                                false);
+                        if (httpHost == null)
+                            return new HttpRoute(target);
+                        else
+                            return new HttpRoute(target, null, httpHost,
+                                    false);
                     } else {
                         LOGGER.debug("Setting https scheme");
-                        return new HttpRoute(target, null, httpsHost,
-                                true);
+                        if (httpsHost == null)
+                            return new HttpRoute(target);
```

Comment:
The HttpRoute constructor accepting a single HttpHost is documented from the java docs to be creating a new direct insecure route. I guess then that when the protocol is https then we should use this other constructor with the secure flag to true:
public HttpRoute([HttpHost](https://hc.apache.org/httpcomponents-core-ga/httpcore/apidocs/org/apache/http/HttpHost.html?is-external=true) target,
[InetAddress](https://download.oracle.com/javase/1.5.0/docs/api/java/net/InetAddress.html?is-external=true) local,
boolean secure);

Reply:
updated the code